docs/design/rule-manager/implementation-details.md

```diff
@@ -0,0 +1,13 @@
+# Implementation details for RuleManager library
+
+Facade of RuleManager lib is implemented as an interface with two methods:
+1. buildRulesForPaths(DataAdapter adapter)
+2. buildRulesForSwitch(DataAdapter adapter)
+
+![Class diagram](class-diagram.png "class diagram")
+
+Different adapter implementations may query database or read a file under the hood. All required parameters are provided as constructor params and filled by the calling component.
+
+To generate default rules generators hierarchy (similar to current SwitchManager implementation) is used but the logic is moved from floodlight to RuleManager lib and result is in openflow-style command format. To generate flow related rules similar way is introduced.
+
+From floodlight perspective 3 new SpeakerCommands added: InstallOfCommand, VerifyOfCommand and RemoveOfCommand. Each command contains only one OfCommand and processed separately. Floodlight should map corresponding OpenFlow-style command into loxigen representation, send it to devices and process results. RuleManager returns collection of OfCommands with dependencies and the calling component (Flow-HS or SwitchManager) is responsible for encapsulating and sending this commands into floodlight in right order.
```

Comment:
This will degrade performance. Why not to send whole batch of commands for specific switch in one request (chain in case of kafka transport)?

Reply:
Reworked.